python/triton/language/standard.py

```diff
@@ -179,7 +179,7 @@ def max(input, axis=None, return_indices=False, return_indices_tie_break_left=Tr
             if core.constexpr(input.dtype.is_floating()):
                 input = input.to(core.float32)
             else:
-                assert input.dtype.is_integer_type()
+                assert core.constexpr(input.dtype.is_int()), "Expecting input to be integer type"
```

Comment:
Why using `core.constexpr`?

Reply:
I figured it I would keep it consistent with the call to is_floating a couple lines above, but it is not necessary so I will remove it